app/assets/javascripts/discourse/tests/unit/lib/case-converter-test.js

```diff
@@ -0,0 +1,32 @@
+import { setupTest } from "ember-qunit";
+import { module, test } from "qunit";
+import {
+  camelCaseToDash,
+  camelCaseToSnakeCase,
+  snakeCaseToCamelCase,
+} from "discourse-common/lib/case-converter";
+
+module("Unit | discourse-common | case-converter", function (hooks) {
+  setupTest(hooks);
+
+  test("camelCaseToSnakeCase", function (assert) {
+    assert.equal(camelCaseToSnakeCase("camelCase"), "camel_case");
+    assert.equal(camelCaseToSnakeCase("camelCase99"), "camel_case99");
+    assert.equal(camelCaseToSnakeCase("camelCaseId"), "camel_case_id");
+    assert.equal(camelCaseToSnakeCase("camelCaseUrl"), "camel_case_url");
```

Comment:
What about trickier cases like `camelCaseURL`?

Reply:
Hmm but supporting it might be tricky though. In Rails, we have to declare the acronyms via some config exposed by Rails.

Reply:
@tgxworld yeah probably more permutations that can be done here but I think in most cases this will be OK; besides this was completely untested previously and is only used in user-search `performSearch` :) Not even sure if I will keep the AdminPlugin the same after `t/132877` internally, will see.